samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                }
+
+                // Command lists are created in the recording state, but there is nothing
+                // to record yet. The main loop expects it to be closed, so close it now.
+                ThrowIfFailed(nameof(ID3D12GraphicsCommandList.Close), _commandList->Close());
+
+                // Create the vertex buffer.
+                {
+                    // Define the geometry for a triangle.
+                    const int TriangleVerticesCount = 3;
+                    var triangleVertices = stackalloc Vertex[TriangleVerticesCount] {
+                        new Vertex {
+                            Position = new Vector3(0.0f, 0.25f * AspectRatio, 0.0f),
+                            Color = new Vector4(1.0f, 0.0f, 0.0f, 1.0f)
+                        },
+                        new Vertex {
+                            Position = new Vector3(0.25f, -0.25f * AspectRatio, 0.0f),
+                            Color = new Vector4(0.0f, 1.0f, 0.0f, 1.0f)
+                        },
+                        new Vertex {
+                            Position = new Vector3(-0.25f, -0.25f * AspectRatio, 0.0f),
+                            Color = new Vector4(0.0f, 0.0f, 1.0f, 1.0f)
+                        },
+                    };
+
+                    var vertexBufferSize = (uint)sizeof(Vertex) * TriangleVerticesCount;
+
+                    // Note: using upload heaps to transfer static data like vert buffers is not
+                    // recommended. Every time the GPU needs it, the upload heap will be marshalled
+                    // over. Please read up on Default Heap usage. An upload heap is used here for
+                    // code simplicity and because there are very few verts to actually transfer.
+                    fixed (ID3D12Resource** vertexBuffer = &_vertexBuffer)
+                    {
+                        var heapProperties = new D3D12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_UPLOAD);
+                        var bufferDesc = D3D12_RESOURCE_DESC.Buffer(vertexBufferSize);
+
+                        iid = IID_ID3D12Resource;
+                        ThrowIfFailed(nameof(ID3D12Device._CreateCommittedResource), _device->CreateCommittedResource(
+                            &heapProperties,
+                            D3D12_HEAP_FLAG_NONE,
+                            &bufferDesc,
+                            D3D12_RESOURCE_STATE_GENERIC_READ,
+                            pOptimizedClearValue: null,
+                            &iid,
+                            (void**)vertexBuffer
+                        ));
+                    }
+
+                    // Copy the triangle data to the vertex buffer.
+                    var readRange = new D3D12_RANGE();
+
+                    byte* pVertexDataBegin;
+                    ThrowIfFailed(nameof(ID3D12Resource._Map), _vertexBuffer->Map(Subresource: 0, &readRange, (void**)&pVertexDataBegin));
```

Comment:
This is probably a mistake in the `HelloTriangle` sample as well:
```suggestion
                    ThrowIfFailed(nameof(ID3D12Resource.Map), _vertexBuffer->Map(Subresource: 0, &readRange, (void**)&pVertexDataBegin));
```